micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -167,7 +167,7 @@ public CountedAspect(MeterRegistry registry, Function<ProceedingJoinPoint, Itera
         this.shouldSkip = shouldSkip;
     }
 
-    @Around("@within(io.micrometer.core.annotation.Counted) and not @annotation(io.micrometer.core.annotation.Counted)")
+    @Around("@within(io.micrometer.core.annotation.Counted) && !@annotation(io.micrometer.core.annotation.Counted) && execution(* *(..))")
```

Comment:
why is there `&& execution(* *(..))` at the end?

Reply:
Because the implementation has an unchecked class-cast to MethodSignature that causes runtime crashes if the pointcut matches a constructor, which it does without this addition that restricts it only to method signatures (with a return type).

* `@within(io.micrometer.core.annotation.Counted)` - will match the methods of class annotated with `@Counted` 
* `&& !@annotation(io.micrometer.core.annotation.Counted)` - excludes methods that are already annotated separately with `@Counted` to avoid double instrumentation
* `&& execution(* *(..))` - matches method signatures only excluding constructors

What happens if you don't include the last part is that if you annotate a class with `@Counted` it will crash at runtime when you try to create a new instance of the class because the pointcut would match the constructor which gives a ConstructorSignature and fails to be cast to MethodSignature in the implementation. The tests cover this, so if you remove that last part you'll see the error when running the tests.